docs/source/format/StatisticsSchema.rst

```diff
@@ -0,0 +1,873 @@
+   * - ``ARROW:max_byte_width:exact``
+     - ``int64``
+     - The maximum size in bytes of a row in the target
+       column. (exact)
+   * - ``ARROW:max_byte_width:approximate``
+     - ``float64``
+     - The maximum size in bytes of a row in the target
+       column. (approximate)
+   * - ``ARROW:max_value:exact``
+     - Target dependent
+     - The maximum value in the target column. (exact)
+   * - ``ARROW:max_value:approximate``
+     - Target dependent
+     - The maximum value in the target column. (approximate)
+   * - ``ARROW:min_value:exact``
+     - Target dependent
+     - The minimum value in the target column. (exact)
+   * - ``ARROW:min_value:approximate``
+     - Target dependent
+     - The minimum value in the target column. (approximate)
+   * - ``ARROW:null_count:exact``
+     - ``int64``
+     - The number of nulls in the target column. (exact)
+   * - ``ARROW:null_count:approximate``
+     - ``float64``
+     - The number of nulls in the target column. (approximate)
+   * - ``ARROW:row_count:exact``
+     - ``int64``
+     - The number of rows in the target table, record batch or
+       array. (exact)
+   * - ``ARROW:row_count:approximate``
+     - ``float64``
+     - The number of rows in the target table, record batch or
+       array. (approximate)
+
+If you find a statistic that might be useful to multiple systems,
+please propose it on the `Apache Arrow development mailing-list
+<https://arrow.apache.org/community/>`__.
+
+Interoperability improves when producers and consumers of statistics
+follow a previously agreed upon statistic specification.
+
+.. _statistics-schema-examples:
+
+Examples
+========
+
+Here are some examples to help you understand.
+
+Simple record batch
+-------------------
+
+Schema::
+
+    vendor_id: int32
+    passenger_count: int64
+
+Data::
+
+    vendor_id:       [5, 1, 5, 1, 5]
+    passenger_count: [1, 1, 2, 0, null]
+
+Statistics:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Target
+     - Name
+     - Value
+   * - Record batch
+     - The number of rows
+     - ``5``
+   * - ``vendor_id``
+     - The number of nulls
+     - ``0``
+   * - ``vendor_id``
+     - The number of distinct values
+     - ``2``
+   * - ``vendor_id``
+     - The max value
+     - ``5``
+   * - ``vendor_id``
+     - The min value
+     - ``1``
+   * - ``passenger_count``
+     - The number of nulls
+     - ``1``
+   * - ``passenger_count``
+     - The number of distinct values
+     - ``3``
+   * - ``passenger_count``
+     - The max value
+     - ``2``
+   * - ``passenger_count``
+     - The min value
+     - ``0``
+
+Column indexes:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Index
+     - Target
+   * - ``0``
+     - ``vendor_id``
+   * - ``1``
+     - ``passenger_count``
+
+Statistics schema::
+
+    struct<
+      column: int32,
+      statistics: map<
+        key: dictionary<values: utf8, indices: int32>,
+        items: dense_union<0: int64>
+      >
+    >
+
+Statistics array::
```

Comment:
Copied from: https://github.com/apache/arrow/pull/43553/files#r1896750644

I didn't understand this example. I thought the statistics were structs, so I would have expected the data to look something like this (perhaps we could give the "logical contents" and then the specific array encoding):

```
[ 
  // first struct element
  { 
    column: null, # record batch
     statistics: {
        "ARROW:row_count:exact": 0
     }
   },
  { 
    column: 0, # vendor_id
     statistics: {
        "ARROW:null_count:exact": 0,
        "ARROW:distinct_count:exact": 2,
        "ARROW:max_value:exact": 5,
        "ARROW:min_value:exact": 1,
     }
   },
...
]
```

I can help work out the example if people think this is a good idea

Reply:
Ah, it makes sense.
I used physical representation (columnar representation) here but logical representation (row based representation) may be easy to understand.

How about showing both of them? (We keep the current representation and add row based representation like you suggested.)